homeassistant/components/switchbot_cloud/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The SwitchBot via API integration."""
+from asyncio import gather
+from dataclasses import dataclass
+from logging import getLogger
+
+from switchbot_api import Device, Remote, SwitchBotAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+
+_LOGGER = getLogger(__name__)
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+@dataclass
+class Data:
+    """Data to use in platforms."""
+
+    api: SwitchBotAPI
+    switches: list[Device | Remote]
+
+
+async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
+    """Set up SwitchBot via API from a config entry."""
+    token = config.data[CONF_API_TOKEN]
+    secret = config.data[CONF_API_KEY]
+
+    api = SwitchBotAPI(token=token, secret=secret)
+    devices = await api.list_devices()
+    _LOGGER.debug("Devices: %s", devices)
+    devices_and_coordinators = [
+        (device, SwitchBotCoordinator(hass, api, device)) for device in devices
+    ]
```

Comment:
I think it makes sense to do all the updates in a single coordinator instead of having multiple, than index the returned data by device id

Reply:
OTOH, the current design allows users to disable devices they don't care about so maybe its better as-is